backend/lib/pix-service/service.js

```diff
@@ -0,0 +1,41 @@
+import { Value } from '@sinclair/typebox/value'
+import { PIECE_INDEXER_URL } from '../config.js'
+import { getMinerPeerId } from '../rpc-service/service.js'
+import { PixResponse } from './data-types.js'
+import pRetry from 'p-retry'
+
+/**
+ * @param {string} providerId
+ * @param {string} pieceCid
+ * @returns {Promise<string>}
+ */
+export const pixRequest = async (providerId, pieceCid) => {
+  const url = PIECE_INDEXER_URL + '/sample/' + providerId + '/' + pieceCid
+  try {
+    const response = await pRetry(async () => await fetch(url, {
+      method: 'GET',
+      headers: { 'content-type': 'application/json' }
+    }), { retries: 5 })
+    const json = await response.json()
+    try {
+      const parsedPixResponse = Value.Parse(PixResponse, json)
+      return parsedPixResponse.samples[0]
+    } catch (e) {
+      e.message = `Failed to parse response from piece indexer: ${e.message}, response: ${JSON.stringify(json)}`
+      throw e
```

Comment:
```suggestion
      throw new Error(`Failed to parse response from piece indexer, response: ${JSON.stringify(json)}`, { cause: err })
```

JS has a native construct for error wrapping now :) https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause

Reply:
Thanks you!